projects/plugins/jetpack/extensions/blocks/conversation/utils.js

```diff
@@ -23,3 +24,181 @@ export function getPlainText( html, escape = false ) {
 
 	return escapeHTML( text );
 }
+
+/**
+ * Return the file extension according to the file name.
+ *
+ * @param {string} filename - file full name.
+ * @returns {string} File extension.
+ */
+export function pickExtensionFromFileName( filename ) {
+	return `.${ filename.substr( filename.lastIndexOf( '.' ) + 1 ) }`;
+}
+
+export const FILE_EXTENSION_SRT = '.srt';
+export const FILE_EXTENSION_TXT = '.txt';
+export const FILE_EXTENSION_VTT = '.vtt';
+export const FILE_EXTENSION_SBV = '.sbv';
+
+export const ACCEPTED_FILE_EXT_ARRAY = [
+	FILE_EXTENSION_SRT,
+	FILE_EXTENSION_TXT,
+	FILE_EXTENSION_VTT,
+	FILE_EXTENSION_SBV,
+];
+
+export const ACCEPTED_FILE_EXTENSIONS = ACCEPTED_FILE_EXT_ARRAY.join( ', ' );
+
+export const TRANSCRIPT_MAX_FILE_SIZE = 100000;
+
+/*
+ * Generic format / template.
+ * ----------------------------
+ * <speaker> <timestamp>
+ *  <content>
+ * ----------------------------
+ * Serices: otter.ai, ...
+ */
+
+const otterRegExp = /(.*[^\s])\s{1,}(\d{1,2}(?::\d{1,2}?)+)\s+\n([\s\S]*?(?=\n{2}|$))/;
+const otterFormatRegExp = new RegExp( otterRegExp, 'gm' );
+const otterFormatTestRegExp = new RegExp( otterRegExp, 'g' );
+
+/*
+ * ----------------------------
+ * <speaker>: [<timestamp>] <content>
+ * ----------------------------
+ * Serices: sonix.ai, ...
+ */
+const sonixRegExp = /(?:(.*[^\s]):\s*)?\[([\d{1,2}:]*)]\s([\s\S]*?(?=\n{1,2}|$))/;
+const sonixFormatRegExp = new RegExp( sonixRegExp, 'gm' );
+const sonixFormatTestRegExp = new RegExp( sonixRegExp, 'g' );
+
+const parsers = [
+	{
+		name: 'otter',
+		re: otterFormatRegExp,
+		testRE: otterFormatTestRegExp,
+	},
+	{
+		name: 'sonix',
+		re: sonixFormatRegExp,
+		testRE: sonixFormatTestRegExp,
+	},
+];
+
+/* SRT format / template.
+ * ----------------------------
+ * <index>
+ * <startTime> --> <endTime>
+ * <content>
+ * ----------------------------
+ * Serices: otter.ai, youtube.com, etc.
+ */
+export const srtFormatRegExp = /(\d+)\n([\d:,]+)\s+-{2}>\s+([\d:,]+)\n([\s\S]*?(?=\n{2}|$))/gm;
+
+export function isAcceptedTranscriptExtension( fileExtension ) {
+	return ACCEPTED_FILE_EXT_ARRAY.indexOf( fileExtension ) >= 0;
+}
+
+export function SRT_parse( content ) {
```

Comment:
It would be great if we had some tests to cover all this.

Reply:
can we do it in a follow-up PR?

Reply:
Sure thing